File: Find closest num to 0 Leetcode 2239/Solution.py
class Solution:
    def findClosestNumber(self, nums: list[int]) -> int:
        closest = nums[0]
        for x in nums:
            if (abs(x) < abs(closest)):
                closest = x
                
        if closest < 0 and abs(closest) in nums:
            return abs(closest)
        else:
            return closest
        
sol = Solution()
nums = [-4,-2,-1,4,8]
print(sol.findClosestNumber(nums))

File: Find closest num to 0 Leetcode 2239/Solution.java
// package Find closest num to 0 Leetcode 2239;

public class Solution {
    public static void main(String[] args) {
        Soln s = new Soln();
        int [] nums = {-4,-2,-1,4,8};
        System.out.println("Hello");
        System.out.println("closest number = "+s.findClosestNumber(nums));
    }
    
}
class Soln {
    public boolean contains(int[] nums, int value){
        for(int c : nums){
            if(c == value){
                return true;
            }
        }
        return false;
    }
    public int findClosestNumber(int[] nums){
        int closest = nums[0];
        for (int x : nums){
            if(Math.abs(x) < Math.abs(closest)){
                closest = x;
            }
        }
        if(closest < 0 && contains(nums, Math.abs(closest))){
            return Math.abs(closest);
        } else {
            return closest;
        }
    }
    
}